lib/OpenQA/WebAPI.pm

```diff
@@ -69,6 +70,24 @@ sub startup {
     # set cookie timeout to 48 hours (will be updated on each request)
     $self->app->sessions->default_expiration(48 * 60 * 60);
 
+    # register YAML output type
+    $self->app->types->type(yaml => 'text/yaml;charset=UTF-8');
+    $self->app->renderer->add_handler(
+        yaml => sub {
+            my ($renderer, $c, $output, $options) = @_;
+            $renderer->{type} = 'yaml';
+            delete $options->{encoding};
+            $$output = YAML::XS::Dump($c->stash->{yaml});
+        });
+    $self->app->hook(
+        before_render => sub {
+            my ($c, $args) = @_;
+            if (exists $args->{yaml} || exists $c->stash->{yaml}) {
+                $args->{format}  = 'yaml';
+                $args->{handler} = 'yaml';
+            }
+        });
```

Comment:
Now just move the handler into a plugin (`OpenQA::WebAPI::Plugin::YAMLRenderer`).